test/MongoUtil_test.py

```diff
@@ -145,18 +147,34 @@ def test_increase_counter_with_multi_threads(self):
         thread_count = 329
 
         threads = list()
+        hids = list()
+        que = queue.Queue()
         for index in range(thread_count):
-            x = threading.Thread(target=mongo_util.increase_counter)
+            x = threading.Thread(target=que.put(mongo_util.increase_counter()))
             threads.append(x)
             x.start()
 
         for index, thread in enumerate(threads):
             thread.join()
 
-        new_counter = mongo_util.get_hid_counter()
+        while not que.empty():
+            hids.append(que.get())
 
+        new_counter = mongo_util.get_hid_counter()
         self.assertEqual(counter + thread_count, new_counter)
 
+        self.assertEqual(len(hids), len(set(hids)))
```

Comment:
check len(hids) = thread_count

Reply:
by I already checked `self.assertEqual(counter + thread_count, new_counter)`

Reply:
This checks that the number of returned items is correct vs. the counter of the returned items in the db. Just being extra conservative

Reply:
👍